x-pack/plugins/enterprise_search/public/applications/shared/schema/add_field_modal/index.tsx

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { ChangeEvent, FormEvent, useEffect, useState } from 'react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { CANCEL_BUTTON_LABEL } from '../../constants';
+import { FIELD_NAME, FIELD_TYPE } from '../constants';
+import { SchemaFieldTypeSelect } from '../index';
+import { SchemaType } from '../types';
+
+import {
+  ADD_FIELD_MODAL_TITLE,
+  ADD_FIELD_MODAL_DESCRIPTION,
+  ADD_FIELD_BUTTON,
+  FORM_ID,
+  FIELD_NAME_PLACEHOLDER,
+  FIELD_NAME_CORRECT_NOTE,
+  FIELD_NAME_CORRECTED_NOTE,
+} from './constants';
+import { formatFieldName } from './utils';
+
+interface Props {
+  addNewField(fieldName: string, newFieldType: string): void;
+  closeAddFieldModal(): void;
+  disableForm?: boolean;
+  addFieldFormErrors?: string[] | null;
+}
+
+export const SchemaAddFieldModal: React.FC<Props> = ({
+  addNewField,
+  addFieldFormErrors,
+  closeAddFieldModal,
+  disableForm,
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [newFieldType, updateNewFieldType] = useState(SchemaType.Text);
+  const [formattedFieldName, setFormattedFieldName] = useState('');
+  const [rawFieldName, setRawFieldName] = useState('');
+
+  useEffect(() => {
+    if (addFieldFormErrors) setLoading(false);
+  }, [addFieldFormErrors]);
+
+  const handleChange = ({ currentTarget: { value } }: ChangeEvent<HTMLInputElement>) => {
+    setRawFieldName(value);
+    setFormattedFieldName(formatFieldName(value));
+  };
+
+  const submitForm = (e: FormEvent) => {
+    e.preventDefault();
+    addNewField(formattedFieldName, newFieldType);
+    setLoading(true);
+  };
+
+  const fieldNameNote =
+    rawFieldName !== formattedFieldName
+      ? FIELD_NAME_CORRECTED_NOTE(formattedFieldName)
+      : FIELD_NAME_CORRECT_NOTE;
+
+  return (
+    <EuiModal onClose={closeAddFieldModal} maxWidth={500}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>{ADD_FIELD_MODAL_TITLE}</EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <p>{ADD_FIELD_MODAL_DESCRIPTION}</p>
+        <EuiSpacer />
+        <EuiForm component="form" id={FORM_ID} onSubmit={submitForm}>
```

Comment:
I think this is registering as a "new" file because of the whitespace changes in https://github.com/elastic/kibana/pull/99096/commits/06314dbebb73ead258e893b50392090178bc403c. Follow along by commit for more granular diffs!